lib/grandpa/grandpa.go

```diff
@@ -854,18 +854,7 @@ func (s *Service) createJustification(bfc common.Hash, stage Subround) ([]Signed
 
 	for _, votes := range eqv {
 		for _, vote := range votes {
-			var signedVote SignedVote = *vote
-			isDescendant, err := s.blockState.IsDescendantOf(bfc, signedVote.Vote.Hash)
-
-			if err != nil {
-				return nil, err
-			}
-
-			if !isDescendant {
-				continue
-			}
-
-			just = append(just, signedVote)
+			just = append(just, *vote)
```

Comment:
Are you sure we can get rid of the isDescandantOf logic here? 🤔

Reply:
@noot, could you explain why we don't need to use `isDescandantOf` here, please?

Reply:
for equivocatory votes, the content of the vote doesn't matter anymore, only the total # of equivocators is what matters for the algorithm. so, it's fine to just append all the equivocatory votes here, then in the verification, it'll just count the # of equivocatory voters.

the exact function that's used is (# votes for a block) + |equivocatory voters| > 2/3 |voters| for a block to be finalized, where |x| denotes # of items in the set. also, in # of votes, votes for a descendant of a block count as a vote for that block as well